doc/languages-frameworks/rust.section.md

```diff
@@ -651,6 +651,66 @@ buildPythonPackage rec {
 }
 ```
 
+#### Rust package built with `meson` {#rust-package-built-with-meson}
+
+Some projects, especially GNOME applications, are built with the Meson Build System instead of calling Cargo directly. Using `rustPlatform.buildRustPackage` may successfully build the main program, but related files will be missing. Instead, you need to set up Cargo dependencies with `fetchCargoTarball` and `cargoSetupHook` and leave the rest to Meson. `rust` and `cargo` are still needed in `nativeBuildInputs` for Meson to use.
+
+```nix
+{ lib
+, stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, rustPlatform
+, rustc
+, cargo
+, wrapGAppsHook4
+, blueprint-compiler
+, libadwaita
+, libsecret
+, tracker
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here. Especially in the documentation.

The `finalAttrs` pattern will let you remove the `rec` keyword ([see its implementation in Nix](https://github.com/NixOS/nix/blob/2d1d81114d72ace89ce08cd3bc93f4eb27a2975d/src/libexpr/nixexpr.cc#L121)).

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.

Reply:
I know about `finalAttrs`. As long as I send a few PRs, I will meet someone asking me to use `finalAttrs`.But regardless of whether finalAttrs is currently used as a standard, the above examples still use rec. Should I change all the above examples?

Reply:
I think the modification of `rec` to `finalAttrs` in the documentation should be done in another PR, after somehow `finalAttrs` is standarized. Until then, I will still write rec to avoid confusion caused by the difference with other examples.

Reply:
I know you know :) It's just that I have predefined replies when doing reviews to speed up the process.

If all the examples uses `rec` maybe it's better to leave it as it is. WDYT @fricklerhandwerk ?

Reply:
Whatever is most likely to succeed and leave users with something usable. Ideally someone shepherds that transition until it's done, otherwise it's likely to lose track or motivation. @DanielSidhion already has an ongoing project with the API format. @Aleksanaa would you be interested in keeping an eye out or driving that?

Reply:
Personally speaking, I'm not interested in it (`finalAttrs`) either. I just follows what nixpkgs is currently having.

Reply:
In that case let's go towards merging this as a strict improvement and leave dealing with `finalAttrs` for later. I don't have capacity for reviews right now though, so I'm fading into the background again. 👋

Reply:
Is there anyone driving the use of `finalAttrs` for Nix code in Nixpkgs as a whole? If yes, I'm working on something for documentation examples which will fit into that. If not, I'll add a note to move documentation examples to using `finalAttrs` whenever possible.